web/source/osk/visualKeyboard.ts

```diff
@@ -1275,10 +1284,16 @@ namespace com.keyman.osk {
 
       // Needs the refreshed layout info to work correctly.
       if(this.currentLayer) {
-        this.currentLayer.refreshLayout(this, paddedHeight, this._computedHeight);
+        this.currentLayer.refreshLayout(this, this._computedHeight - this.getVerticalLayerGroupPadding());
       }
     }
 
+    private getVerticalLayerGroupPadding(): number {
+      // For touch-based OSK layouts, kmwosk.css may include top & bottom padding on the layer-group element.
+      const computedGroupStyle = getComputedStyle(this.layerGroup.element);
+      return parseInt(computedGroupStyle.paddingTop) + parseInt(computedGroupStyle.paddingBottom);
```

Comment:
```suggestion
      return parseInt(computedGroupStyle.paddingTop, 10) + parseInt(computedGroupStyle.paddingBottom, 10);
```

`parseInt` without radix prior to 2021 [was bad](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#octal_interpretations_with_no_radix). Even after 2021 it's still [not good](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#description)